services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluRequest.kt

```diff
@@ -0,0 +1,12 @@
+package fi.hel.haitaton.hanke.tormaystarkastelu
+
+import java.time.ZonedDateTime
+import org.geojson.FeatureCollection
+
+data class TormaystarkasteluRequest(
+    val geometriat: FeatureCollection,
```

Comment:
Since the data structure for e.g. project area is like this:
```
{
   "id":58,
   "hankeId":10,
   "haittaAlkuPvm":"2024-05-23T00:00:00Z",
   "haittaLoppuPvm":"2024-05-26T00:00:00Z",
   "geometriat":{
      "id":59,
      "featureCollection":{
         "type":"FeatureCollection",
         "crs":{
            "type":"name",
            "properties":{
               "name":"urn:ogc:def:crs:EPSG::3879"
            }
         },
         "features":[
            {
               "type":"Feature",
               "properties":{
                  "hankeTunnus":"HAI24-10"
               },
               "geometry":{
                  "type":"Polygon",
                  "crs":{
                     "type":"name",
                     "properties":{
                        "name":"EPSG:3879"
                     }
                  },
                  "coordinates":[
                     [
                        [
                           25494576.42,
                           6683925.71
                        ],
                        [
                           25494565.16,
                           6683915.33
                        ],
                        [
                           25494560.24,
                           6683921.13
                        ],
                        [
                           25494565.2,
                           6683925.56
                        ],
                        [
                           25494571.67,
                           6683931.34
                        ],
                        [
                           25494576.42,
                           6683925.71
                        ]
                     ]
                  ]
               }
            }
         ]
      },
      "version":51,
      "createdByUserId":"5f893af3-f7c0-433b-bf50-08f88fbc43a7",
      "createdAt":"2024-05-23T11:43:44.483Z",
      "modifiedByUserId":"5f893af3-f7c0-433b-bf50-08f88fbc43a7",
      "modifiedAt":"2024-05-31T07:48:37.156Z"
   },
   "kaistaHaitta":"VAHENTAA_SAMANAIKAISESTI_KAISTAN_KAHDELLA_AJOSUUNNALLA",
   "kaistaPituusHaitta":"PITUUS_ALLE_10_METRIA",
   "meluHaitta":"SATUNNAINEN_MELUHAITTA",
   "polyHaitta":"TOISTUVA_POLYHAITTA",
   "tarinaHaitta":"JATKUVA_TARINAHAITTA",
   "nimi":"Hankealue 2",
   "tormaystarkasteluTulos":{
      "autoliikenneindeksi":1.3,
      "pyoraliikenneindeksi":3,
      "linjaautoliikenneindeksi":0,
      "raitioliikenneindeksi":0,
      "liikennehaittaindeksi":{
         "indeksi":3,
         "tyyppi":"PYORALIIKENNEINDEKSI"
      }
   },
   "haittojenhallintasuunnitelma":{
      "PYORALIIKENNE":"",
      "YLEINEN":"alueen 2 yleinen haittojenhallintasuunnitelma",
      "AUTOLIIKENNE":"",
      "LINJAAUTOLIIKENNE":"",
      "MUUT":"",
      "RAITIOLIIKENNE":""
   }
}
```

I was wondering whether it would be better to name this `geometriat` field as `featureCollection` instead? Because `geometriat` is the parent data structure in project area API.

Reply:
Good point. I took it a bit further and changed the request structure to match the hankealue object, so that the UI can use the hankealue JSON as a request without any modifications.
```
{
  "geometriat": {
    "featureCollection": <GeoJSON FeatureCollection -objekti>
  },
  "haittaAlkuPvm" : "2025-02-20T23:45:56Z",
  "haittaLoppuPvm" : "2025-02-21T00:12:34Z",
  "kaistaHaitta" : "VAHENTAA_KAISTAN_YHDELLA_AJOSUUNNALLA",
  "kaistaPituusHaitta" : "PITUUS_10_99_METRIA"
}
```